app/src/components/pages/ScheduleManage.tsx

```diff
@@ -1,7 +1,202 @@
+        Manage Existing Schedules
+      </Typography>
+      <Grid container rowSpacing={3} alignItems={'center'}>
+        <Grid item>
+          <TextField
+            id="outlined-basic"
+            label="Search schedules..."
+            variant="outlined"
+            type="search"
+            value={search}
+            onChange={(e) => filter(e.target.value)}
+            className="input"
+            placeholder="Keyword"
+            sx={{ width: 350 }}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <List
+            sx={{
+              width: '100%',
+              bgcolor: 'background.paper',
+              position: 'relative',
+              overflow: 'auto',
+              maxHeight: 350,
+              '& ul': { padding: 0 }
+            }}
+            subheader={<li />}
+          >
+            {foundSchedules && foundSchedules.length > 0 ? (
+              foundSchedules.map((schedule) => (
+                <ListItem key={schedule.id}>
+                  <ListItemAvatar>
+                    <Avatar sx={{ bgcolor: getBgColor(schedule.term) }}>{getIcon(schedule.term)}</Avatar>
+                  </ListItemAvatar>
+                  <ListItemText primary={schedule.term} secondary={schedule.year} />
+                  <Button variant="outlined">View</Button>
+                  <Typography variant="body2" marginLeft={2}>
+                    Generated on {schedule.createdAt.toString()}
+                  </Typography>
+                </ListItem>
+              ))
+            ) : (
+              <Typography variant="subtitle1">No schedules found!</Typography>
+            )}
+          </List>
+        </Grid>
+
+        <Typography variant="h5" marginY={4}>
+          Generate New Schedule
+        </Typography>
+        <Grid container spacing={2} alignItems={'center'}>
+          <Grid item>
+            <TextField
+              id="select-semester"
+              select
+              label="Select Semester"
+              value={semester}
+              onChange={(e) => setSemester(e.target.value)}
+              style={{ width: '300px' }}
+            >
+              {semesterTypes.map((semester) => (
+                <MenuItem key={semester} value={semester}>
+                  {semester}
+                </MenuItem>
+              ))}
+            </TextField>
+          </Grid>
+          <Grid item>
+            <LocalizationProvider dateAdapter={AdapterDateFns}>
+              <DatePicker
+                views={['year']}
+                label="Year"
+                value={year}
+                onChange={(e) => setYear(e)}
+                renderInput={(params) => <TextField {...params} helperText={null} />}
+              />
+            </LocalizationProvider>
+          </Grid>
+          <Grid item>
+            <Button
+              variant="contained"
+              size="large"
+              onClick={() =>
+                navigate(`/schedule/generate`, {

```

Comment:
I don't think navigating is neccessary

Reply:
Why not? We want this to redirect to the generate page and this passes the state through nicely. Do you think a `window.href` set would be better or am I misunderstanding what you're saying

Reply:
I think i misunderstood the workflow, I thought it would lead to a loading page or the same page with a loading notif when we integrate. Going to the generate page works too